crates/chia-consensus/src/merkle_tree.rs

```diff
@@ -631,6 +652,33 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_bad_proofs_2() {
+        // Create a random number generator
+        let mut rng = SmallRng::seed_from_u64(1337);
+        // Generate a random length for the Vec
+        let vec_length: usize = rng.gen_range(5..=500);
+
+        // Generate a Vec of random [u8; 32] arrays
+        let mut random_data: Vec<[u8; 32]> = Vec::with_capacity(vec_length);
+
+        let mut array: [u8; 32] = [0; 32];
+        rng.fill(&mut array);
+        random_data.push(array);
+
+        let mut bad_proof: Vec<u8> = Vec::new();
+        bad_proof.push(MIDDLE);
+        bad_proof.push(TRUNCATED);
+        bad_proof.extend_from_slice(&random_data[0]);
+        bad_proof.push(MIDDLE);
+        bad_proof.push(TERMINAL);
+        bad_proof.extend_from_slice(&[0x1; 32]); // this ought to be on the right
```

Comment:
I don't think this comment is right. The two leafs you have start with `0b00000000` and `0b00000001`. i.e. they share the first 7 bits. The proof is expected to have 8 layers. This is a valid test (that proofs must be complete), but I don't think it tests what you meant to test.